services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -0,0 +1,965 @@
+    )
+
+    logging.info(f"get config names for {dataset=}")
+    previous_step = "/config-names"
+    try:
+        response = get_response(kind=previous_step, dataset=dataset)
+    except DoesNotExist as e:
+        raise DatasetNotFoundError(f"No response found in previous step '{previous_step}' for this dataset.", e) from e
+    if response["http_status"] != HTTPStatus.OK:
+        raise PreviousStepStatusError(f"Previous step {previous_step} gave an error: {response['http_status']}..")
+
+    config_names_content = response["content"]
+    if "config_names" not in config_names_content:
+        raise PreviousStepFormatError("Previous step did not return the expected content: 'config_names'.")
+
+    if not isinstance(config_names_content["config_names"], list):
+        raise PreviousStepFormatError(
+            "Previous step did not return the expected content.",
+            TypeError(f"config_names should be a list, but got {type(config_names_content['config_names'])}"),
+        )
+
+    config_names = {config_name_item["config"] for config_name_item in config_names_content["config_names"]}
+    if config not in config_names:
+        raise ConfigNamesError(f"{config=} does not exist in {dataset=}")
+
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+
+    # prepare the parquet files locally
+    local_parquet_files: List[ParquetFile] = []
+    download_config = DownloadConfig(delete_extracted=True)
+    try:
+        builder = load_dataset_builder(
+            path=dataset,
+            name=config,
+            revision=source_revision,
+            use_auth_token=hf_token,
+            download_config=download_config,
+        )
+    except _EmptyDatasetError as err:
+        raise EmptyDatasetError(f"{dataset=} is empty.", cause=err) from err
+    raise_if_too_big_from_external_data_files(
+        builder=builder,
+        max_dataset_size=max_dataset_size,
+        max_external_data_files=max_external_data_files,
+        hf_token=hf_token,
+    )
+    builder.download_and_prepare(file_format="parquet")  # the parquet files are stored in the cache dir
+    dataset_info = asdict(builder.info)
+    local_parquet_files.extend(
+        ParquetFile(local_file=local_file, local_dir=builder.cache_dir, config=config)
+        for local_file in glob.glob(f"{builder.cache_dir}**/*.parquet")
+    )
+
+    # create the target revision if it does not exist yet (clone from initial commit to avoid cloning all repo's files)
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+        if all(ref.ref != target_revision for ref in refs.converts):
+            initial_commit = hf_api.list_repo_commits(repo_id=dataset, repo_type=DATASET_TYPE)[-1].commit_id
+            committer_hf_api.create_branch(
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=initial_commit
```

Comment:
Note that this closes https://github.com/huggingface/datasets-server/issues/846!